hiop +rocm build fails: g++-11: unrecognized command-line option '--offload-arch=gfx90a'

### Steps to reproduce the issue

`hiop +rocm` build is failing using:
* `spack@develop` (5e014be8b7aaecfec5d98aaa406dd6393610942b from `Tue Apr 19 16:48:49 2022 -0400`)
* Ubuntu 20.04, x86_64
* GCC 11.1.0
* ROCm 4.5.2

Concrete spec: [hiop.spec.json.txt](https://github.com/spack/spack/files/8517895/hiop.spec.json.txt)

Concretization:
```
hiop@0.5.3%gcc@11.1.0~cuda~deepchecking~ipo~jsrun~kron+mpi~raja+rocm~shared~sparse amdgpu_target=gfx90a build_type=RelWithDebInfo arch=linux-ubuntu20.04-x86_64
    ^cmake@3.23.1%gcc@11.1.0~doc+ncurses+ownlibs~qt build_type=Release arch=linux-ubuntu20.04-x86_64
        ^ncurses@6.2%gcc@11.1.0~symlinks+termlib abi=none arch=linux-ubuntu20.04-x86_64
            ^pkgconf@1.8.0%gcc@11.1.0 arch=linux-ubuntu20.04-x86_64
        ^openssl@1.1.1n%gcc@11.1.0~docs~shared certs=system arch=linux-ubuntu20.04-x86_64
            ^perl@5.34.1%gcc@11.1.0+cpanm+shared+threads arch=linux-ubuntu20.04-x86_64
                ^berkeley-db@18.1.40%gcc@11.1.0+cxx~docs+stl patches=b231fcc arch=linux-ubuntu20.04-x86_64
                ^bzip2@1.0.8%gcc@11.1.0~debug~pic+shared arch=linux-ubuntu20.04-x86_64
                    ^diffutils@3.8%gcc@11.1.0 arch=linux-ubuntu20.04-x86_64
                        ^libiconv@1.16%gcc@11.1.0 libs=shared,static arch=linux-ubuntu20.04-x86_64
                ^gdbm@1.19%gcc@11.1.0 arch=linux-ubuntu20.04-x86_64
                    ^readline@8.1%gcc@11.1.0 arch=linux-ubuntu20.04-x86_64
                ^zlib@1.2.12%gcc@11.1.0+optimize+pic+shared patches=0d38234 arch=linux-ubuntu20.04-x86_64
    ^hip@4.5.2%gcc@11.1.0~ipo build_type=Release patches=24ca25a,2a41904 arch=linux-ubuntu20.04-x86_64
    ^hipblas@4.5.2%gcc@11.1.0~ipo build_type=Release patches=a2cc151 arch=linux-ubuntu20.04-x86_64
    ^hipsparse@4.5.2%gcc@11.1.0~ipo build_type=Release arch=linux-ubuntu20.04-x86_64
    ^hsa-rocr-dev@4.5.2%gcc@11.1.0+image~ipo+shared build_type=Release patches=71e6851 arch=linux-ubuntu20.04-x86_64
    ^llvm-amdgpu@4.5.2%gcc@11.1.0~ipo~link_llvm_dylib~llvm_dylib~openmp+rocm-device-libs build_type=Release patches=d999f3b arch=linux-ubuntu20.04-x86_64
    ^magma@2.6.1%gcc@11.1.0~cuda+fortran~ipo+rocm+shared amdgpu_target=gfx1011 build_type=RelWithDebInfo arch=linux-ubuntu20.04-x86_64
        ^openblas@0.3.20%gcc@11.1.0~bignuma~consistent_fpcsr~ilp64+locking+pic+shared symbol_suffix=none threads=openmp arch=linux-ubuntu20.04-x86_64
    ^mpich@4.0.2%gcc@11.1.0~argobots+fortran+hwloc+hydra+libxml2+pci+romio~slurm~two_level_namespace~verbs~wrapperrpath device=ch4 netmod=ofi pmi=pmi arch=linux-ubuntu20.04-x86_64
        ^findutils@4.8.0%gcc@11.1.0 patches=440b954 arch=linux-ubuntu20.04-x86_64
        ^hwloc@2.7.1%gcc@11.1.0~cairo~cuda~gl~libudev+libxml2~netloc~nvml~opencl+pci~rocm+shared arch=linux-ubuntu20.04-x86_64
            ^libpciaccess@0.16%gcc@11.1.0 arch=linux-ubuntu20.04-x86_64
                ^libtool@2.4.7%gcc@11.1.0 arch=linux-ubuntu20.04-x86_64
                    ^m4@1.4.19%gcc@11.1.0+sigsegv patches=9dc5fbd,bfdffa7 arch=linux-ubuntu20.04-x86_64
                        ^libsigsegv@2.13%gcc@11.1.0 arch=linux-ubuntu20.04-x86_64
                ^util-macros@1.19.3%gcc@11.1.0 arch=linux-ubuntu20.04-x86_64
            ^libxml2@2.9.12%gcc@11.1.0~python arch=linux-ubuntu20.04-x86_64
                ^xz@5.2.5%gcc@11.1.0+pic libs=shared,static arch=linux-ubuntu20.04-x86_64
        ^libfabric@1.14.0%gcc@11.1.0~debug~kdreg fabrics=rxm,sockets,tcp,udp arch=linux-ubuntu20.04-x86_64
```
 
Build issue easily reproduced using cached dependency binaries using the following:
* Docker container image: `ecpe4s/ubuntu20.04-rocm:4.5.2`
  * Contains ROCm 4.5.2 installed @ `/opt/rocm-4.5.2`
* Spack environment: [spack.yaml.txt](https://github.com/spack/spack/files/8517970/spack.yaml.txt)
  * Uses ROCm 4.5.2 as externals defined in `packages:` section

To reproduce:

```
$> docker run -it ecpe4s/ubuntu20.04-rocm:4.5.2

root@7d183a0138f2:/# git clone https://github.com/spack/spack
root@7d183a0138f2:/# (cd spack && git checkout 5e014be8b7)
root@7d183a0138f2:/# . spack/share/spack/setup-env.sh

root@7d183a0138f2:/# spack mirror add E4S https://cache.e4s.io/hiop
root@7d183a0138f2:/# spack buildcache keys -it

root@7d183a0138f2:/# spack -e . install --only dependencies --include-build-deps --cache-only
# Very quick install using cached dependencies, <3 minutes on a fast connection

root@7d183a0138f2:/# spack -e . install --no-cache
...
==> Installing hiop-0.5.3-m5gce25zokjf7dnrmcogbngeanqlwg67
==> No binary for hiop-0.5.3-m5gce25zokjf7dnrmcogbngeanqlwg67 found: installing from source
==> No patches needed for hiop
==> hiop: Executing phase: 'cmake'
==> hiop: Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2:
    'make' '-j16'

40 errors found in build log:
...
     123    cd /tmp/root/spack-stage/spack-stage-hiop-0.5.3-m5gce25zokjf7dnrmcogbngeanqlwg67/spack-build-m5gce25/src/Optimization && /spack/lib/spack/env/gcc/g++ -DHAVE_HIP -D__HI
            P_PLATFORM_AMD__=1 -D__HIP_PLATFORM_HCC__=1 -I/tmp/root/spack-stage/spack-stage-hiop-0.5.3-m5gce25zokjf7dnrmcogbngeanqlwg67/spack-build-m5gce25 -I/tmp/root/spack-stage
            /spack-stage-hiop-0.5.3-m5gce25zokjf7dnrmcogbngeanqlwg67/spack-src/src/Interface -I/tmp/root/spack-stage/spack-stage-hiop-0.5.3-m5gce25zokjf7dnrmcogbngeanqlwg67/spack-
            src/src/Optimization -I/tmp/root/spack-stage/spack-stage-hiop-0.5.3-m5gce25zokjf7dnrmcogbngeanqlwg67/spack-src/src/LinAlg -I/tmp/root/spack-stage/spack-stage-hiop-0.5.
            3-m5gce25zokjf7dnrmcogbngeanqlwg67/spack-src/src/Utils -I/tmp/root/spack-stage/spack-stage-hiop-0.5.3-m5gce25zokjf7dnrmcogbngeanqlwg67/spack-src/src/_Internals -I/spac
            k/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/magma-2.6.1-ixglz2etllcmgrd4pyrjfivv5b6uvu4p/include -isystem /spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/mpic
            h-4.0.2-pgzwxngke6u7ecvnbm5jmihtqnyi27da/include -isystem /opt/rocm-4.5.2/hip/../include -isystem /opt/rocm-4.5.2/llvm/lib/clang/13.0.0/include/.. -isystem /opt/rocm-4
            .5.2/hip/include -isystem /opt/rocm-4.5.2/include -O2 -g -DNDEBUG -fPIC -x hip --offload-arch=gfx90a -std=gnu++11 -MD -MT src/Optimization/CMakeFiles/hiopOptimization.
            dir/hiopAlgFilterIPM.cpp.o -MF CMakeFiles/hiopOptimization.dir/hiopAlgFilterIPM.cpp.o.d -o CMakeFiles/hiopOptimization.dir/hiopAlgFilterIPM.cpp.o -c /tmp/root/spack-st
            age/spack-stage-hiop-0.5.3-m5gce25zokjf7dnrmcogbngeanqlwg67/spack-src/src/Optimization/hiopAlgFilterIPM.cpp
  >> 124    g++-11: error: unrecognized command-line option '--offload-arch=gfx90a'
  >> 125    g++-11: error: unrecognized command-line option '--offload-arch=gfx90a'
  >> 126    make[2]: *** [src/Utils/CMakeFiles/hiopUtils.dir/build.make:93: src/Utils/CMakeFiles/hiopUtils.dir/hiopOptions.cpp.o] Error 1
     127    make[2]: *** Waiting for unfinished jobs....
  >> 128    make[2]: *** [src/LinAlg/CMakeFiles/hiopLinAlg.dir/build.make:79: src/LinAlg/CMakeFiles/hiopLinAlg.dir/hiopVectorPar.cpp.o] Error 1
```

### Error message

<details><summary>Error message</summary><pre>
...
==> Installing hiop-0.5.3-m5gce25zokjf7dnrmcogbngeanqlwg67
==> No binary for hiop-0.5.3-m5gce25zokjf7dnrmcogbngeanqlwg67 found: installing from source
==> No patches needed for hiop
==> hiop: Executing phase: 'cmake'
==> hiop: Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2:
    'make' '-j16'

40 errors found in build log:
     118    cd /tmp/root/spack-stage/spack-stage-hiop-0.5.3-m5gce25zokjf7dnrmcogbngeanqlwg67/spack-build-m5gce25/src/Optimization && /spack/lib/spack/env/gcc/g++ -DHAVE_HIP -D__HI
            P_PLATFORM_AMD__=1 -D__HIP_PLATFORM_HCC__=1 -I/tmp/root/spack-stage/spack-stage-hiop-0.5.3-m5gce25zokjf7dnrmcogbngeanqlwg67/spack-build-m5gce25 -I/tmp/root/spack-stage
            /spack-stage-hiop-0.5.3-m5gce25zokjf7dnrmcogbngeanqlwg67/spack-src/src/Interface -I/tmp/root/spack-stage/spack-stage-hiop-0.5.3-m5gce25zokjf7dnrmcogbngeanqlwg67/spack-
            src/src/Optimization -I/tmp/root/spack-stage/spack-stage-hiop-0.5.3-m5gce25zokjf7dnrmcogbngeanqlwg67/spack-src/src/LinAlg -I/tmp/root/spack-stage/spack-stage-hiop-0.5.
            3-m5gce25zokjf7dnrmcogbngeanqlwg67/spack-src/src/Utils -I/tmp/root/spack-stage/spack-stage-hiop-0.5.3-m5gce25zokjf7dnrmcogbngeanqlwg67/spack-src/src/_Internals -I/spac
            k/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/magma-2.6.1-ixglz2etllcmgrd4pyrjfivv5b6uvu4p/include -isystem /spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/mpic
            h-4.0.2-pgzwxngke6u7ecvnbm5jmihtqnyi27da/include -isystem /opt/rocm-4.5.2/hip/../include -isystem /opt/rocm-4.5.2/llvm/lib/clang/13.0.0/include/.. -isystem /opt/rocm-4
            .5.2/hip/include -isystem /opt/rocm-4.5.2/include -O2 -g -DNDEBUG -fPIC -x hip --offload-arch=gfx90a -std=gnu++11 -MD -MT src/Optimization/CMakeFiles/hiopOptimization.
            dir/hiopFilter.cpp.o -MF CMakeFiles/hiopOptimization.dir/hiopFilter.cpp.o.d -o CMakeFiles/hiopOptimization.dir/hiopFilter.cpp.o -c /tmp/root/spack-stage/spack-stage-hi
            op-0.5.3-m5gce25zokjf7dnrmcogbngeanqlwg67/spack-src/src/Optimization/hiopFilter.cpp
     119    cd /tmp/root/spack-stage/spack-stage-hiop-0.5.3-m5gce25zokjf7dnrmcogbngeanqlwg67/spack-build-m5gce25/src/Optimization && /spack/lib/spack/env/gcc/g++ -DHAVE_HIP -D__HI
            P_PLATFORM_AMD__=1 -D__HIP_PLATFORM_HCC__=1 -I/tmp/root/spack-stage/spack-stage-hiop-0.5.3-m5gce25zokjf7dnrmcogbngeanqlwg67/spack-build-m5gce25 -I/tmp/root/spack-stage
            /spack-stage-hiop-0.5.3-m5gce25zokjf7dnrmcogbngeanqlwg67/spack-src/src/Interface -I/tmp/root/spack-stage/spack-stage-hiop-0.5.3-m5gce25zokjf7dnrmcogbngeanqlwg67/spack-
            src/src/Optimization -I/tmp/root/spack-stage/spack-stage-hiop-0.5.3-m5gce25zokjf7dnrmcogbngeanqlwg67/spack-src/src/LinAlg -I/tmp/root/spack-stage/spack-stage-hiop-0.5.
            3-m5gce25zokjf7dnrmcogbngeanqlwg67/spack-src/src/Utils -I/tmp/root/spack-stage/spack-stage-hiop-0.5.3-m5gce25zokjf7dnrmcogbngeanqlwg67/spack-src/src/_Internals -I/spac
            k/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/magma-2.6.1-ixglz2etllcmgrd4pyrjfivv5b6uvu4p/include -isystem /spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/mpic
            h-4.0.2-pgzwxngke6u7ecvnbm5jmihtqnyi27da/include -isystem /opt/rocm-4.5.2/hip/../include -isystem /opt/rocm-4.5.2/llvm/lib/clang/13.0.0/include/.. -isystem /opt/rocm-4
            .5.2/hip/include -isystem /opt/rocm-4.5.2/include -O2 -g -DNDEBUG -fPIC -x hip --offload-arch=gfx90a -std=gnu++11 -MD -MT src/Optimization/CMakeFiles/hiopOptimization.
            dir/hiopIterate.cpp.o -MF CMakeFiles/hiopOptimization.dir/hiopIterate.cpp.o.d -o CMakeFiles/hiopOptimization.dir/hiopIterate.cpp.o -c /tmp/root/spack-stage/spack-stage
            -hiop-0.5.3-m5gce25zokjf7dnrmcogbngeanqlwg67/spack-src/src/Optimization/hiopIterate.cpp
     120    cd /tmp/root/spack-stage/spack-stage-hiop-0.5.3-m5gce25zokjf7dnrmcogbngeanqlwg67/spack-build-m5gce25/src/LinAlg && /spack/lib/spack/env/gcc/g++ -DHAVE_HIP -D__HIP_PLAT
            FORM_AMD__=1 -D__HIP_PLATFORM_HCC__=1 -I/tmp/root/spack-stage/spack-stage-hiop-0.5.3-m5gce25zokjf7dnrmcogbngeanqlwg67/spack-build-m5gce25 -I/tmp/root/spack-stage/spack
            -stage-hiop-0.5.3-m5gce25zokjf7dnrmcogbngeanqlwg67/spack-src/src/Interface -I/tmp/root/spack-stage/spack-stage-hiop-0.5.3-m5gce25zokjf7dnrmcogbngeanqlwg67/spack-src/sr
            c/Optimization -I/tmp/root/spack-stage/spack-stage-hiop-0.5.3-m5gce25zokjf7dnrmcogbngeanqlwg67/spack-src/src/LinAlg -I/tmp/root/spack-stage/spack-stage-hiop-0.5.3-m5gc
            e25zokjf7dnrmcogbngeanqlwg67/spack-src/src/Utils -I/tmp/root/spack-stage/spack-stage-hiop-0.5.3-m5gce25zokjf7dnrmcogbngeanqlwg67/spack-src/src/_Internals -I/spack/opt/
            spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/magma-2.6.1-ixglz2etllcmgrd4pyrjfivv5b6uvu4p/include -isystem /spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/mpich-4.0.
            2-pgzwxngke6u7ecvnbm5jmihtqnyi27da/include -isystem /opt/rocm-4.5.2/hip/../include -isystem /opt/rocm-4.5.2/llvm/lib/clang/13.0.0/include/.. -isystem /opt/rocm-4.5.2/h
            ip/include -isystem /opt/rocm-4.5.2/include -O2 -g -DNDEBUG -fPIC -x hip --offload-arch=gfx90a -std=gnu++11 -MD -MT src/LinAlg/CMakeFiles/hiopLinAlg.dir/hiopLinAlgFact
            ory.cpp.o -MF CMakeFiles/hiopLinAlg.dir/hiopLinAlgFactory.cpp.o.d -o CMakeFiles/hiopLinAlg.dir/hiopLinAlgFactory.cpp.o -c /tmp/root/spack-stage/spack-stage-hiop-0.5.3-
            m5gce25zokjf7dnrmcogbngeanqlwg67/spack-src/src/LinAlg/hiopLinAlgFactory.cpp
     121    [ 27%] Building CXX object src/Optimization/CMakeFiles/hiopOptimization.dir/hiopAlgFilterIPM.cpp.o
     122    cd /tmp/root/spack-stage/spack-stage-hiop-0.5.3-m5gce25zokjf7dnrmcogbngeanqlwg67/spack-build-m5gce25/src/Optimization && /spack/lib/spack/env/gcc/g++ -DHAVE_HIP -D__HI
            P_PLATFORM_AMD__=1 -D__HIP_PLATFORM_HCC__=1 -I/tmp/root/spack-stage/spack-stage-hiop-0.5.3-m5gce25zokjf7dnrmcogbngeanqlwg67/spack-build-m5gce25 -I/tmp/root/spack-stage
            /spack-stage-hiop-0.5.3-m5gce25zokjf7dnrmcogbngeanqlwg67/spack-src/src/Interface -I/tmp/root/spack-stage/spack-stage-hiop-0.5.3-m5gce25zokjf7dnrmcogbngeanqlwg67/spack-
            src/src/Optimization -I/tmp/root/spack-stage/spack-stage-hiop-0.5.3-m5gce25zokjf7dnrmcogbngeanqlwg67/spack-src/src/LinAlg -I/tmp/root/spack-stage/spack-stage-hiop-0.5.
            3-m5gce25zokjf7dnrmcogbngeanqlwg67/spack-src/src/Utils -I/tmp/root/spack-stage/spack-stage-hiop-0.5.3-m5gce25zokjf7dnrmcogbngeanqlwg67/spack-src/src/_Internals -I/spac
            k/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/magma-2.6.1-ixglz2etllcmgrd4pyrjfivv5b6uvu4p/include -isystem /spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/mpic
            h-4.0.2-pgzwxngke6u7ecvnbm5jmihtqnyi27da/include -isystem /opt/rocm-4.5.2/hip/../include -isystem /opt/rocm-4.5.2/llvm/lib/clang/13.0.0/include/.. -isystem /opt/rocm-4
            .5.2/hip/include -isystem /opt/rocm-4.5.2/include -O2 -g -DNDEBUG -fPIC -x hip --offload-arch=gfx90a -std=gnu++11 -MD -MT src/Optimization/CMakeFiles/hiopOptimization.
            dir/hiopResidual.cpp.o -MF CMakeFiles/hiopOptimization.dir/hiopResidual.cpp.o.d -o CMakeFiles/hiopOptimization.dir/hiopResidual.cpp.o -c /tmp/root/spack-stage/spack-st
            age-hiop-0.5.3-m5gce25zokjf7dnrmcogbngeanqlwg67/spack-src/src/Optimization/hiopResidual.cpp
     123    cd /tmp/root/spack-stage/spack-stage-hiop-0.5.3-m5gce25zokjf7dnrmcogbngeanqlwg67/spack-build-m5gce25/src/Optimization && /spack/lib/spack/env/gcc/g++ -DHAVE_HIP -D__HI
            P_PLATFORM_AMD__=1 -D__HIP_PLATFORM_HCC__=1 -I/tmp/root/spack-stage/spack-stage-hiop-0.5.3-m5gce25zokjf7dnrmcogbngeanqlwg67/spack-build-m5gce25 -I/tmp/root/spack-stage
            /spack-stage-hiop-0.5.3-m5gce25zokjf7dnrmcogbngeanqlwg67/spack-src/src/Interface -I/tmp/root/spack-stage/spack-stage-hiop-0.5.3-m5gce25zokjf7dnrmcogbngeanqlwg67/spack-
            src/src/Optimization -I/tmp/root/spack-stage/spack-stage-hiop-0.5.3-m5gce25zokjf7dnrmcogbngeanqlwg67/spack-src/src/LinAlg -I/tmp/root/spack-stage/spack-stage-hiop-0.5.
            3-m5gce25zokjf7dnrmcogbngeanqlwg67/spack-src/src/Utils -I/tmp/root/spack-stage/spack-stage-hiop-0.5.3-m5gce25zokjf7dnrmcogbngeanqlwg67/spack-src/src/_Internals -I/spac
            k/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/magma-2.6.1-ixglz2etllcmgrd4pyrjfivv5b6uvu4p/include -isystem /spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/mpic
            h-4.0.2-pgzwxngke6u7ecvnbm5jmihtqnyi27da/include -isystem /opt/rocm-4.5.2/hip/../include -isystem /opt/rocm-4.5.2/llvm/lib/clang/13.0.0/include/.. -isystem /opt/rocm-4
            .5.2/hip/include -isystem /opt/rocm-4.5.2/include -O2 -g -DNDEBUG -fPIC -x hip --offload-arch=gfx90a -std=gnu++11 -MD -MT src/Optimization/CMakeFiles/hiopOptimization.
            dir/hiopAlgFilterIPM.cpp.o -MF CMakeFiles/hiopOptimization.dir/hiopAlgFilterIPM.cpp.o.d -o CMakeFiles/hiopOptimization.dir/hiopAlgFilterIPM.cpp.o -c /tmp/root/spack-st
            age/spack-stage-hiop-0.5.3-m5gce25zokjf7dnrmcogbngeanqlwg67/spack-src/src/Optimization/hiopAlgFilterIPM.cpp
  >> 124    g++-11: error: unrecognized command-line option '--offload-arch=gfx90a'
  >> 125    g++-11: error: unrecognized command-line option '--offload-arch=gfx90a'
  >> 126    make[2]: *** [src/Utils/CMakeFiles/hiopUtils.dir/build.make:93: src/Utils/CMakeFiles/hiopUtils.dir/hiopOptions.cpp.o] Error 1
     127    make[2]: *** Waiting for unfinished jobs....
  >> 128    make[2]: *** [src/LinAlg/CMakeFiles/hiopLinAlg.dir/build.make:79: src/LinAlg/CMakeFiles/hiopLinAlg.dir/hiopVectorPar.cpp.o] Error 1
     129    make[2]: *** Waiting for unfinished jobs....
  >> 130    g++-11: error: unrecognized command-line option '--offload-arch=gfx90a'
  >> 131    make[2]: *** [src/Utils/CMakeFiles/hiopUtils.dir/build.make:79: src/Utils/CMakeFiles/hiopUtils.dir/hiopLogger.cpp.o] Error 1
     132    make[2]: Leaving directory '/tmp/root/spack-stage/spack-stage-hiop-0.5.3-m5gce25zokjf7dnrmcogbngeanqlwg67/spack-build-m5gce25'
  >> 133    make[1]: *** [CMakeFiles/Makefile2:333: src/Utils/CMakeFiles/hiopUtils.dir/all] Error 2
     134    make[1]: *** Waiting for unfinished jobs....
  >> 135    g++-11: error: unrecognized command-line option '--offload-arch=gfx90a'
  >> 136    make[2]: *** [src/Optimization/CMakeFiles/hiopOptimization.dir/build.make:135: src/Optimization/CMakeFiles/hiopOptimization.dir/hiopFilter.cpp.o] Error 1
     137    make[2]: *** Waiting for unfinished jobs....
  >> 138    g++-11: error: unrecognized command-line option '--offload-arch=gfx90a'
  >> 139    make[2]: *** [src/Optimization/CMakeFiles/hiopOptimization.dir/build.make:107: src/Optimization/CMakeFiles/hiopOptimization.dir/hiopIterate.cpp.o] Error 1
  >> 140    g++-11: error: unrecognized command-line option '--offload-arch=gfx90a'
     141    [ 29%] Building CXX object src/Optimization/CMakeFiles/hiopOptimization.dir/hiopKKTLinSysMDS.cpp.o
  >> 142    g++-11: error: unrecognized command-line option '--offload-arch=gfx90a'
  >> 143    make[2]: *** [src/Optimization/CMakeFiles/hiopOptimization.dir/build.make:93: src/Optimization/CMakeFiles/hiopOptimization.dir/hiopFactAcceptor.cpp.o] Error 1
     144    cd /tmp/root/spack-stage/spack-stage-hiop-0.5.3-m5gce25zokjf7dnrmcogbngeanqlwg67/spack-build-m5gce25/src/Optimization && /spack/lib/spack/env/gcc/g++ -DHAVE_HIP -D__HI
            P_PLATFORM_AMD__=1 -D__HIP_PLATFORM_HCC__=1 -I/tmp/root/spack-stage/spack-stage-hiop-0.5.3-m5gce25zokjf7dnrmcogbngeanqlwg67/spack-build-m5gce25 -I/tmp/root/spack-stage
            /spack-stage-hiop-0.5.3-m5gce25zokjf7dnrmcogbngeanqlwg67/spack-src/src/Interface -I/tmp/root/spack-stage/spack-stage-hiop-0.5.3-m5gce25zokjf7dnrmcogbngeanqlwg67/spack-
            src/src/Optimization -I/tmp/root/spack-stage/spack-stage-hiop-0.5.3-m5gce25zokjf7dnrmcogbngeanqlwg67/spack-src/src/LinAlg -I/tmp/root/spack-stage/spack-stage-hiop-0.5.
            3-m5gce25zokjf7dnrmcogbngeanqlwg67/spack-src/src/Utils -I/tmp/root/spack-stage/spack-stage-hiop-0.5.3-m5gce25zokjf7dnrmcogbngeanqlwg67/spack-src/src/_Internals -I/spac
            k/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/magma-2.6.1-ixglz2etllcmgrd4pyrjfivv5b6uvu4p/include -isystem /spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/mpic
            h-4.0.2-pgzwxngke6u7ecvnbm5jmihtqnyi27da/include -isystem /opt/rocm-4.5.2/hip/../include -isystem /opt/rocm-4.5.2/llvm/lib/clang/13.0.0/include/.. -isystem /opt/rocm-4
            .5.2/hip/include -isystem /opt/rocm-4.5.2/include -O2 -g -DNDEBUG -fPIC -x hip --offload-arch=gfx90a -std=gnu++11 -MD -MT src/Optimization/CMakeFiles/hiopOptimization.
            dir/hiopKKTLinSysMDS.cpp.o -MF CMakeFiles/hiopOptimization.dir/hiopKKTLinSysMDS.cpp.o.d -o CMakeFiles/hiopOptimization.dir/hiopKKTLinSysMDS.cpp.o -c /tmp/root/spack-st
            age/spack-stage-hiop-0.5.3-m5gce25zokjf7dnrmcogbngeanqlwg67/spack-src/src/Optimization/hiopKKTLinSysMDS.cpp
  >> 145    make[2]: *** [src/Optimization/CMakeFiles/hiopOptimization.dir/build.make:149: src/Optimization/CMakeFiles/hiopOptimization.dir/hiopAlgFilterIPM.cpp.o] Error 1
     146    [ 31%] Building CXX object src/Optimization/CMakeFiles/hiopOptimization.dir/hiopKKTLinSys.cpp.o
  >> 147    g++-11: error: unrecognized command-line option '--offload-arch=gfx90a'
     148    cd /tmp/root/spack-stage/spack-stage-hiop-0.5.3-m5gce25zokjf7dnrmcogbngeanqlwg67/spack-build-m5gce25/src/Optimization && /spack/lib/spack/env/gcc/g++ -DHAVE_HIP -D__HI
            P_PLATFORM_AMD__=1 -D__HIP_PLATFORM_HCC__=1 -I/tmp/root/spack-stage/spack-stage-hiop-0.5.3-m5gce25zokjf7dnrmcogbngeanqlwg67/spack-build-m5gce25 -I/tmp/root/spack-stage
            /spack-stage-hiop-0.5.3-m5gce25zokjf7dnrmcogbngeanqlwg67/spack-src/src/Interface -I/tmp/root/spack-stage/spack-stage-hiop-0.5.3-m5gce25zokjf7dnrmcogbngeanqlwg67/spack-
            src/src/Optimization -I/tmp/root/spack-stage/spack-stage-hiop-0.5.3-m5gce25zokjf7dnrmcogbngeanqlwg67/spack-src/src/LinAlg -I/tmp/root/spack-stage/spack-stage-hiop-0.5.
            3-m5gce25zokjf7dnrmcogbngeanqlwg67/spack-src/src/Utils -I/tmp/root/spack-stage/spack-stage-hiop-0.5.3-m5gce25zokjf7dnrmcogbngeanqlwg67/spack-src/src/_Internals -I/spac
            k/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/magma-2.6.1-ixglz2etllcmgrd4pyrjfivv5b6uvu4p/include -isystem /spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/mpic
            h-4.0.2-pgzwxngke6u7ecvnbm5jmihtqnyi27da/include -isystem /opt/rocm-4.5.2/hip/../include -isystem /opt/rocm-4.5.2/llvm/lib/clang/13.0.0/include/.. -isystem /opt/rocm-4
            .5.2/hip/include -isystem /opt/rocm-4.5.2/include -O2 -g -DNDEBUG -fPIC -x hip --offload-arch=gfx90a -std=gnu++11 -MD -MT src/Optimization/CMakeFiles/hiopOptimization.
            dir/hiopKKTLinSys.cpp.o -MF CMakeFiles/hiopOptimization.dir/hiopKKTLinSys.cpp.o.d -o CMakeFiles/hiopOptimization.dir/hiopKKTLinSys.cpp.o -c /tmp/root/spack-stage/spack
            -stage-hiop-0.5.3-m5gce25zokjf7dnrmcogbngeanqlwg67/spack-src/src/Optimization/hiopKKTLinSys.cpp
     149    [ 32%] Building CXX object src/Optimization/CMakeFiles/hiopOptimization.dir/hiopHessianLowRank.cpp.o
  >> 150    make[2]: *** [src/LinAlg/CMakeFiles/hiopLinAlg.dir/build.make:149: src/LinAlg/CMakeFiles/hiopLinAlg.dir/hiopMatrixMDS.cpp.o] Error 1
     151    cd /tmp/root/spack-stage/spack-stage-hiop-0.5.3-m5gce25zokjf7dnrmcogbngeanqlwg67/spack-build-m5gce25/src/Optimization && /spack/lib/spack/env/gcc/g++ -DHAVE_HIP -D__HI
            P_PLATFORM_AMD__=1 -D__HIP_PLATFORM_HCC__=1 -I/tmp/root/spack-stage/spack-stage-hiop-0.5.3-m5gce25zokjf7dnrmcogbngeanqlwg67/spack-build-m5gce25 -I/tmp/root/spack-stage
            /spack-stage-hiop-0.5.3-m5gce25zokjf7dnrmcogbngeanqlwg67/spack-src/src/Interface -I/tmp/root/spack-stage/spack-stage-hiop-0.5.3-m5gce25zokjf7dnrmcogbngeanqlwg67/spack-
            src/src/Optimization -I/tmp/root/spack-stage/spack-stage-hiop-0.5.3-m5gce25zokjf7dnrmcogbngeanqlwg67/spack-src/src/LinAlg -I/tmp/root/spack-stage/spack-stage-hiop-0.5.
            3-m5gce25zokjf7dnrmcogbngeanqlwg67/spack-src/src/Utils -I/tmp/root/spack-stage/spack-stage-hiop-0.5.3-m5gce25zokjf7dnrmcogbngeanqlwg67/spack-src/src/_Internals -I/spac
            k/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/magma-2.6.1-ixglz2etllcmgrd4pyrjfivv5b6uvu4p/include -isystem /spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/mpic
            h-4.0.2-pgzwxngke6u7ecvnbm5jmihtqnyi27da/include -isystem /opt/rocm-4.5.2/hip/../include -isystem /opt/rocm-4.5.2/llvm/lib/clang/13.0.0/include/.. -isystem /opt/rocm-4
            .5.2/hip/include -isystem /opt/rocm-4.5.2/include -O2 -g -DNDEBUG -fPIC -x hip --offload-arch=gfx90a -std=gnu++11 -MD -MT src/Optimization/CMakeFiles/hiopOptimization.
            dir/hiopHessianLowRank.cpp.o -MF CMakeFiles/hiopOptimization.dir/hiopHessianLowRank.cpp.o.d -o CMakeFiles/hiopOptimization.dir/hiopHessianLowRank.cpp.o -c /tmp/root/sp
            ack-stage/spack-stage-hiop-0.5.3-m5gce25zokjf7dnrmcogbngeanqlwg67/spack-src/src/Optimization/hiopHessianLowRank.cpp
  >> 152    g++-11: error: unrecognized command-line option '--offload-arch=gfx90a'
  >> 153    make[2]: *** [src/Optimization/CMakeFiles/hiopOptimization.dir/build.make:121: src/Optimization/CMakeFiles/hiopOptimization.dir/hiopResidual.cpp.o] Error 1
  >> 154    g++-11: error: unrecognized command-line option '--offload-arch=gfx90a'
  >> 155    make[2]: *** [src/LinAlg/CMakeFiles/hiopLinAlg.dir/build.make:107: src/LinAlg/CMakeFiles/hiopLinAlg.dir/hiopMatrixDenseRowMajor.cpp.o] Error 1
  >> 156    g++-11: error: unrecognized command-line option '--offload-arch=gfx90a'
  >> 157    make[2]: *** [src/LinAlg/CMakeFiles/hiopLinAlg.dir/build.make:163: src/LinAlg/CMakeFiles/hiopLinAlg.dir/hiopMatrixComplexDense.cpp.o] Error 1
  >> 158    g++-11: error: unrecognized command-line option '--offload-arch=gfx90a'
  >> 159    g++-11: error: unrecognized command-line option '--offload-arch=gfx90a'
  >> 160    make[2]: *** [src/LinAlg/CMakeFiles/hiopLinAlg.dir/build.make:93: src/LinAlg/CMakeFiles/hiopLinAlg.dir/hiopVectorIntSeq.cpp.o] Error 1
  >> 161    g++-11: error: unrecognized command-line option '--offload-arch=gfx90a'
  >> 162    make[2]: *** [src/Optimization/CMakeFiles/hiopOptimization.dir/build.make:79: src/Optimization/CMakeFiles/hiopOptimization.dir/hiopNlpFormulation.cpp.o] Error 1
  >> 163    g++-11: error: unrecognized command-line option '--offload-arch=gfx90a'
  >> 164    make[2]: *** [src/LinAlg/CMakeFiles/hiopLinAlg.dir/build.make:135: src/LinAlg/CMakeFiles/hiopLinAlg.dir/hiopLinAlgFactory.cpp.o] Error 1
  >> 165    make[2]: *** [src/LinAlg/CMakeFiles/hiopLinAlg.dir/build.make:121: src/LinAlg/CMakeFiles/hiopLinAlg.dir/hiopLinSolver.cpp.o] Error 1
     166    make[2]: Leaving directory '/tmp/root/spack-stage/spack-stage-hiop-0.5.3-m5gce25zokjf7dnrmcogbngeanqlwg67/spack-build-m5gce25'
  >> 167    make[1]: *** [CMakeFiles/Makefile2:307: src/LinAlg/CMakeFiles/hiopLinAlg.dir/all] Error 2
  >> 168    g++-11: error: unrecognized command-line option '--offload-arch=gfx90a'
  >> 169    make[2]: *** [src/Optimization/CMakeFiles/hiopOptimization.dir/build.make:177: src/Optimization/CMakeFiles/hiopOptimization.dir/hiopKKTLinSysMDS.cpp.o] Error 1
  >> 170    g++-11: error: unrecognized command-line option '--offload-arch=gfx90a'
  >> 171    make[2]: *** [src/Optimization/CMakeFiles/hiopOptimization.dir/build.make:191: src/Optimization/CMakeFiles/hiopOptimization.dir/hiopHessianLowRank.cpp.o] Error 1
  >> 172    g++-11: error: unrecognized command-line option '--offload-arch=gfx90a'
  >> 173    make[2]: *** [src/Optimization/CMakeFiles/hiopOptimization.dir/build.make:163: src/Optimization/CMakeFiles/hiopOptimization.dir/hiopKKTLinSys.cpp.o] Error 1
     174    make[2]: Leaving directory '/tmp/root/spack-stage/spack-stage-hiop-0.5.3-m5gce25zokjf7dnrmcogbngeanqlwg67/spack-build-m5gce25'
  >> 175    make[1]: *** [CMakeFiles/Makefile2:281: src/Optimization/CMakeFiles/hiopOptimization.dir/all] Error 2
     176    make[2]: Leaving directory '/tmp/root/spack-stage/spack-stage-hiop-0.5.3-m5gce25zokjf7dnrmcogbngeanqlwg67/spack-build-m5gce25'
     177    [ 32%] Built target hiopInterface
     178    make[1]: Leaving directory '/tmp/root/spack-stage/spack-stage-hiop-0.5.3-m5gce25zokjf7dnrmcogbngeanqlwg67/spack-build-m5gce25'
  >> 179    make: *** [Makefile:149: all] Error 2

See build log for details:
  /tmp/root/spack-stage/spack-stage-hiop-0.5.3-m5gce25zokjf7dnrmcogbngeanqlwg67/spack-build-out.txt
...
</pre></details>


### Information on your system

* **Spack:** 0.18.0.dev0 (5e014be8b7aaecfec5d98aaa406dd6393610942b)
* **Python:** 3.8.10
* **Platform:** linux-ubuntu20.04-cascadelake
* **Concretizer:** clingo

### Additional information

[spack-build-env.txt](https://github.com/spack/spack/files/8517953/spack-build-env.txt)
[spack-build-out.txt](https://github.com/spack/spack/files/8517954/spack-build-out.txt)

@ashermancinelli 
@CameronRutherford 

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

--- a/var/spack/repos/builtin/packages/hiop/package.py
+++ b/var/spack/repos/builtin/packages/hiop/package.py
@@ -115,6 +115,9 @@ class Hiop(CMakePackage, CudaPackage, ROCmPackage):
                 self.define('HIOP_MAGMA_DIR', spec['magma'].prefix),
             ])
 
+        if '+rocm' in spec:
+            args.append(self.define('CMAKE_CXX_COMPILER', spec['hip'].hipcc))
+
         args.extend([
             self.define('HIOP_BUILD_STATIC', True),
             self.define('LAPACK_FOUND', True),

--- a/var/spack/repos/builtin/packages/hiop/package.py
+++ b/var/spack/repos/builtin/packages/hiop/package.py
@@ -115,9 +115,6 @@ class Hiop(CMakePackage, CudaPackage, ROCmPackage):
                 self.define('HIOP_MAGMA_DIR', spec['magma'].prefix),
             ])
 
-        if '+rocm' in spec:
-            args.append(self.define('CMAKE_CXX_COMPILER', spec['hip'].hipcc))
-
         args.extend([
             self.define('HIOP_BUILD_STATIC', True),
             self.define('LAPACK_FOUND', True),
@@ -172,6 +169,8 @@ class Hiop(CMakePackage, CudaPackage, ROCmPackage):
         #     self.define('HIP_CLANG_INCLUDE_PATH',
         #         '/opt/rocm-X.Y.Z/llvm/lib/clang/14.0.0/include/'))
         if '+rocm' in spec:
+            args.append(self.define('CMAKE_CXX_COMPILER', spec['hip'].hipcc))
+
             rocm_arch_list = spec.variants['amdgpu_target'].value
             if rocm_arch_list[0] != 'none':
                 args.append(self.define('GPU_TARGETS', rocm_arch_list))

